client/state/sites/selectors.js

```diff
@@ -337,6 +340,21 @@ export function isSitePreviewable( state, siteId ) {
 	return !! unmappedUrl && isHttps( unmappedUrl );
 }
 
+/**
+ * Returns true if current user can purchase upgrades for this site
+ *
+ * @param  {Object}   state  Global state tree
+ * @param  {Number}   siteId Site ID
+ * @return {?Boolean}        Whether site is previewable
+ */
+export function canCurrentUserUpgradeSite( state, siteId = null ) {
+	if ( ! siteId ) {
+		siteId = getSelectedSiteId( state );
+	}
+	const isPaid = isCurrentPlanPaid( state, siteId );
+	return ! isPaid || isCurrentUserCurrentPlanOwner( state, siteId );
```

Comment:
The user also needs to have the manage_options capability to buy upgrades. I'm not sure if I understand the function properly but if I do, an additional check may be needed just like you do in `canCurrentUserUseAds`